http-clients/java-nio-client/src/main/java/software/amazon/awssdk/http/nio/java/internal/JavaHttpRequestFactory.java

```diff
@@ -43,11 +43,12 @@ private String getRequestMethod(SdkHttpMethod sdkhttpmethod) {
         return sdkhttpmethod.name();
     }
 
-    private HttpRequest.BodyPublisher createBodyPublisher(SdkHttpContentPublisher sdkHttpContentPublisher) {
+    public static HttpRequest.BodyPublisher createBodyPublisher(AsyncExecuteRequest asyncExecuteRequest) {
```

Comment:
Why is this public?

Reply:
This was changed into public to make it testable, probably protected should be better?